Refactor edition

facundovictor opened the pull request:
- Update the util function "get_content_from_editor" to be able to
load the item content to the editor if any.
- Update the cabinet wrapper by adding the function "get_item_content".
This function allows to get the items content without printing anything
to the stdout unless there is some error.
- Remove the -u option from the subcommand "add". Updating should
have its own subcommand.
- Add the subcommand "edit" with support for loading the current item content
within the default editor.

facundovictor committed:
Update get_content_from_editor to load content

Update the util function "get_content_from_editor" to be able to
load the item content if any.

facundovictor committed:
Update wrapper to allow returning an item content

Update the cabinet wrapper by adding the function "get_item_content".
This function allows to get the items content without printing anything
to the stdout unless there is some error.

facundovictor committed:
Remove the -u option from "add"

Remove the -u option from the subcommand "add". Updating should
have its own subcommand.

facundovictor committed:
Add the subcommand "edit"

Add the "edit" subcommand with support for loading the current content
of an item to the editor.

ivanalejandro0 commented on cli.py:
```diff
@@ -65,10 +63,47 @@ def add(ctx, name, tags, content, from_stdin, editor, update):
+              help="Use the default editor for entering the content")
+@click.pass_context
+def update(ctx, name, tags, content, from_stdin, editor):
+    """Add an item to cabinet"""
```
copy/paste leftover :) ... this is for edit/update.

facundovictor replied:
Ops.. yes.. copy/paste. Fixing right away!

ivanalejandro0 reviewed (changes requested):
Cool, I was waiting for a separation between `add`/`update` :).

Regarding the update, if the user doesn't specify tags when updates the contents, the tags are cleared up. We should preserve the existing tags if no `--tag` switch is used.
We also should allow the user to update only the tags and leave the content as it is, but let's tackle that in a different PR :)

Regarding docstrings, we should document using python's types like `str` or `dict` instead of `String` or `Dictionary`, it has the same informative value and uses the exact expected type name.

facundovictor committed:
Set proper description for subcommand update

facundovictor committed:
Add item existence checking before updating

Add checks for only try to update an item once its existence is
confirmed.

facundovictor committed:
Use python's types within docstrings

Update docstrings to use python's types instead of a long
capitalized version of the type description.

facundovictor commented:
Thanks for the review!!! =D
Regarding how we should handle updating tags, yes, we should discuss different approaches and it should be addressed in a different PR.
For now, I've updated the docstrings for using the python's types. And, I've updated the description of the sub-command "update".
